modules/drivers/bigquery-cloud-sdk/src/metabase/driver/bigquery_cloud_sdk.clj

```diff
@@ -58,6 +59,73 @@
                   (.setCredentials (.createScoped creds bigquery-scopes)))]
     (.. bq-bldr build getService)))
 
+;;; +----------------------------------------------------------------------------------------------------------------+
+;;; |                                         Transducing Query Results                                              |
+;;; +----------------------------------------------------------------------------------------------------------------+
+(def ^:private ^:dynamic *page-callback*
+  "Callback to execute when a new page is retrieved, used for testing"
+  nil)
+
+(defn- values-iterator [^TableResult page]
+  (.iterator (.getValues page)))
+
+(defn- reducible-bigquery
```

Comment:
Another nitpick but something like `reducible-results` would be clearer here. It's already clear where talking about BigQuery since where in the BigQuery driver code, so that's redundant. But it would be nice to specify what's reducible here -- "Reducible <insert database engine>" could really mean anything, e.g. who knows if `reducible-postgres`is talking about JDBC result sets or synced metadata or what